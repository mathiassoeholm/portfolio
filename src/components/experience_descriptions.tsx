export interface ExperienceDescription {
  title: string,
  fullName: string,
  period: string,
  description: string[],
}

const experienceDescriptions:{ [index:string] : ExperienceDescription } = {
  'Mindfox': {
    title: 'Co-founder & Software Engineer',
    fullName: 'Mindfox Studios',
    period: 'Jun. 2018 - Present',
    description: [
      'Shipped two original mobile games, Space Loops and Sausage Throw.',
      'Worked with React Native to implement Odin ICE-Pro, from design to release in just three weeks.\n',
      'Mobile development consultancy for LEGO.',
      'Other clients include Kamstrup, AKQA, Novo Nordisk and Destination Aarhus.',
    ],
  },
  'LEGO': {
    title: 'Mobile developer consultant',
    fullName: 'LEGO Group',
    period: 'Sep. 2018 - Jan. 2019',
    description: [
      'Worked with two teams on the apps: LEGO® Movie Maker and LEGO® TV.',
      'Wrote Kotlin for Android, utilizing coroutines to create synchrounous code that’s easy to read.\n',
      'Wrote Swift for iOS, making good use of PromiseKit to implement an advanced share-video flow.',
      'Only engineer to work on both iOS and Android.',
    ],
  },
  'BitBreak': {
    title: 'Co-Founder & Software Engineer',
    fullName: 'BitBreak',
    period: 'Nov. 2016 - Jun. 2018',
    description: [
      'Released a VR game for HTC Vive (In Your Face TD) with 98% positive reviews on Steam (77 positive, 1 negative).',
      'Created an online cryptocurrency vendor with React and Redux.',
      'Did Unity consulting for the game Pool Legends and publisher Spil Games.',
      'All of this was while studying Computer Science at Aarhus University.',
    ],
  },
  'Kiloo': {
    title: 'Software Engineer',
    fullName: 'Kiloo',
    period: 'Apr. 2013 - Jun 2016',
    description: [
      'Worked on a large production (Stormblades) as one of only two programmers. Implemented the entire UI and all meta features, including leaderboards and cloud synchronization.',
      'Made tools in Unity which are now used in one of the most downloaded mobile games ever, Subway Surfers.',
    ],
  },
  'Campus Dj.': {
    title: 'Workshop Instructor',
    fullName: 'Campus Djursland',
    period: 'Aug. 2012 - Apr. 2013',
    description: [
      'Teached game development in C# and Unity, with classes of up to 60 students.',
    ],
  },
  'ClearCut': {
    title: 'Co-Founder & Programmer',
    fullName: 'ClearCut Games',
    period: 'Mar. 2011 - Apr. 2013',
    description: [
      'Released the mobile game Cado for iOS',
      'Wrote C# code day and night, while keeping up with school ;-)',
    ],
  },
};

export default experienceDescriptions;
