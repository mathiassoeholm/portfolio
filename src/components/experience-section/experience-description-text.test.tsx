import React from 'react'
import { cleanup, render, wait } from "react-testing-library"
import ExperienceDescriptionText from "./experience-description-text"
import { ExperienceDescription } from "./experience_descriptions"

describe('experience-description-text', () => {

  it('renders the text set by setText', async () => {
    let setText: (experienceDescription: ExperienceDescription) => void = () => {}

    const { container } = render(
      <ExperienceDescriptionText
        setSetText={(s) => setText = s}
        delayOverride={jest.fn()}
      />
    )

    setText({
      title: 'Founder/Software Engineer',
      fullName: 'Mindfox Studios',
      period: 'Jun. 2018 - Present',
      description: [
        'Shipped two original mobile games, Space Loops and Sausage Throw.',
        'Worked with React Native to implement Odin ICE-Pro, from design to release in just three weeks.\n',
        'Mobile development consultancy for LEGO.',
        'Other clients include Stibo, Kamstrup, AKQA, Novo Nordisk and Destination Aarhus.',
      ],
    })

    // This will allow setText to finish
    await wait()

    expect(container.firstChild).toMatchSnapshot()

    cleanup()
  })
})
