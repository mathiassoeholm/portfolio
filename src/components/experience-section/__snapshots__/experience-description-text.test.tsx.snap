// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`experience-description-text renders the text set by setText 1`] = `
<div
  class="pf-experience-text"
>
  <p
    class="pf-title"
  >
    Founder/Software Engineer
  </p>
  <p>
    @ 
    Mindfox Studios
  </p>
  <p
    class="pf-period"
  >
    Jun. 2018 - Present
  </p>
  <ul
    class="pf-description"
  >
    <li
      style="opacity: 0.15972222222222232; transform: translateY(42.013888888888886px) translateZ(0);"
    >
      Shipped two original mobile games, Space Loops and Sausage Throw.
    </li>
    <li
      style="opacity: 0; transform: translateY(50px) translateZ(0);"
    >
      Worked with React Native to implement Odin ICE-Pro, from design to release in just three weeks.

    </li>
    <li
      style="opacity: 0; transform: translateY(50px) translateZ(0);"
    >
      Mobile development consultancy for LEGO.
    </li>
    <li
      style="opacity: 0; transform: translateY(50px) translateZ(0);"
    >
      Other clients include Stibo, Kamstrup, AKQA, Novo Nordisk and Destination Aarhus.
    </li>
  </ul>
</div>
`;
